utils/token.go

```diff
@@ -0,0 +1,49 @@
+package utils
+
+import (
+	"io/ioutil"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+// GenerateAstarteJWTFromKeyFile generates an Astarte Token for a specific API out of a Private Key File
+func GenerateAstarteJWTFromKeyFile(privateKeyFile string, astarteService AstarteService,
+	authorizationClaims []string, expiry int64) (jwtString string, err error) {
+	keyPEM, err := ioutil.ReadFile(privateKeyFile)
+	if err != nil {
+		return "", err
+	}
+
+	return GenerateAstarteJWTFromPEMKey(keyPEM, astarteService, authorizationClaims, expiry)
+}
+
+// GenerateAstarteJWTFromPEMKey generates an Astarte Token for a specific API out of a Private Key PEM bytearray
+func GenerateAstarteJWTFromPEMKey(privateKeyPEM []byte, astarteService AstarteService,
+	authorizationClaims []string, expiry int64) (jwtString string, err error) {
+	key, err := jwt.ParseRSAPrivateKeyFromPEM(privateKeyPEM)
+	if err != nil {
+		return "", err
+	}
+
+	accessClaimKey := astarteService.JwtClaim()
+
+	if authorizationClaims == nil {
+		authorizationClaims = []string{"^.*$::^.*$"}
+	}
+
+	now := time.Now().UTC().Unix()
+	exp := now + expiry
+	token := jwt.NewWithClaims(jwt.SigningMethodRS256, jwt.MapClaims{
+		accessClaimKey: authorizationClaims,
+		"iat":          now,
+		"exp":          exp,
```

Comment:
`exp` goes in the claims only if `expiry > 0`, so you can generate infinite tokens